src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/BlockPlacer.java

```diff
@@ -53,6 +57,23 @@ public AutonomousMachineHandler getItemHandler() {
 					}
 					else {
 						block.setType(e.getItem().getType());
+						if(e.getItem().hasItemMeta() && (e.getItem().getItemMeta() instanceof BlockStateMeta)) {
```

Comment:
Thanks! Did not know that getItemMeta() will never return null.

Reply:
Well I think you should keep the hasItemName() for that exact reason.

Reply:
> Well I think you should keep the hasItemName() for that exact reason.

Although hasItemMeta() is redundant because getItemMeta() never returns null?